mindsdb/integrations/libs/api_handler_generator.py

```diff
@@ -0,0 +1,535 @@
+        """
+        paths = self.openapi_spec_parser.get_paths()
+        schemas = self.openapi_spec_parser.get_schemas()
+        self.security_schemes = self.openapi_spec_parser.get_security_schemes()
+
+        self.resource_types = self.process_resource_types(schemas)
+        endpoints = self.process_endpoints(paths)
+
+        prefix_len = len(find_common_url_prefix([i.url for i in endpoints]))
+
+        for endpoint in endpoints:
+            url = endpoint.url[prefix_len:]
+            # replace placehoders with x
+            url = re.sub(r"{(\w+)}", 'x', url)
+            url = url.replace('/', '_').strip('_')
+            table_name = table_name_format.format(url=url, method=endpoint.method).lower()
+            self.resources[table_name] = RestApiTable(handler, endpoint=endpoint, resource_gen=self)
+
+        return self.resources
+
+    def process_resource_types(self, schemas: dict) -> dict:
+        resource_types = {}
+        for name, schema_info in schemas.items():
+            resource_types[name] = self._convert_to_resource_type(schema_info)
+
+        return resource_types
+
+    def process_endpoints(self, paths: dict) -> List[APIEndpoint]:
+        """
+        Processes the endpoints defined in the OpenAPI specification.
+
+        Args:
+            endpoints (Dict): A dictionary containing the endpoints defined in the OpenAPI specification.
+
+        Returns:
+            Dict: A dictionary containing the processed endpoints.
+        """
+        endpoints = []
+        for path, path_info in paths.items():
+            if self.api_base is not None and not path.startswith(self.api_base):
+                continue
+
+            for http_method, method_info in path_info.items():
+                if http_method != 'get':
+                    continue
+
+                parameters = self._process_endpoint_parameters(method_info['parameters']) if 'parameters' in method_info else {}
+
+                response = self._process_endpoint_response(method_info['responses'])
+                if response['type'] is None:
+                    continue
+
+                endpoint = APIEndpoint(
+                    url=path,
+                    method=http_method,
+                    params=parameters,
+                    response=response
+                )
+
+                endpoints.append(endpoint)
+
+        return endpoints
+
+    def _process_endpoint_parameters(self, parameters: list) -> Dict[str, APIEndpointParam]:
+        """
+        Processes the parameters defined in the OpenAPI specification.
+
+        Args:
+            parameters (Dict): A dictionary containing the parameters defined in the OpenAPI specification.
+
+        Returns:
+            Dict: A dictionary containing the processed parameters.
+        """
+        endpoint_parameters = {}
+        for parameter in parameters:
+
+            type_name = self.get_resource_type(parameter['schema'])
```

Comment:
Maybe use parametert.get('schema')

Reply:
get_resource_type has to have something on input to return type.
For github api: sometimes there is no 'schema' key, but '$ref' exists and it is possible to find schema using it